apps/src/blockly/addons/contextMenu.ts

```diff
@@ -386,21 +460,75 @@ function setAllWorkspacesTheme(
-  // Remove some default context menu options, if they are present.
   // This needs to be wrapped in a try for now because our GoogleBlocklyWrapperTest.js
   // is not correctly cleaning up its state.
   try {
+    // Option to collapse or expand a block.
     GoogleBlockly.ContextMenuRegistry.registry.unregister(
       'blockCollapseExpand'
     );
+    // Option to open help for a block. Overrided to use our documentation.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('blockHelp');
+    // Option to use inline inputs .
     GoogleBlockly.ContextMenuRegistry.registry.unregister('blockInline');
-    // cleanUp() doesn't currently account for immovable blocks.
+    // Option to clean up blocks. cleanUp() doesn't currently account for immovable blocks.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('cleanWorkspace');
+    // Option to collapse all blocks on a workspace.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('collapseWorkspace');
+    // Option to expand all blocks on a workspace.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('expandWorkspace');
+    // Option to delete all blocks on a workspace.
     GoogleBlockly.ContextMenuRegistry.registry.unregister('workspaceDelete');
   } catch (error) {}
 }
+
+function registerCustomBlockOptions() {
+  // Each option has a "weight": a number that determines the sort order of the option.
+  // Options with higher weights appear later in the context menu.
+
+  // Expected registered block options (from Core and plugins):
+  // 0: blockCopyToStorage
+  // 2: blockComment
+  // 5: blockDisable
+  // 6: blockDelete
+
+  // Our custom options should show below the registered default options.
+  let nextWeight = 7;
+
+  // Custom block options. We increment the weight for each so they are automatically
+  // sorted in the order listed here.
+  registerHelp(nextWeight++);
+  registerDeletable(nextWeight++);
+  registerMovable(nextWeight++);
+  registerEditable(nextWeight++);
+  registerShadow(nextWeight++);
+  registerUnshadow(nextWeight);
+}
+
+function registerCustomWorkspaceOptions() {
+  // Each option has a "weight": a number that determines the sort order of the option.
+  // Options with higher weights appear later in the context menu.
+
+  // Expected registered workspace options (from Core and plugins):
+  // 0: blockPasteFromStorage
+  // 1: undoWorkspace
+  // 2: redoWorkspace
+
+  // Our custom options should show below the registered default options.
+  let nextWeight = 3;
+
+  // Custom workspace options. We increment the weight for each so they are automatically
+  // sorted in the order listed here. Cursors and Themes have multiple options so we need to adjust
+  // the weight accordingly.
+  registerKeyboardNavigation(nextWeight++);
+  registerAllCursors(nextWeight, NAVIGATION_CURSOR_TYPES);
+  nextWeight += NAVIGATION_CURSOR_TYPES.length;
+  registerDarkMode(nextWeight++);
+  registerThemes(nextWeight, themes);
```

Comment:
should this be `nextWeight++`?

Reply:
Not needed. The `++` incrementation happens after the value is accessed so that it's ready for the next option to be registered. This is the last option, so we don't need to increment at all. The comment below this line explains that we'd want to count up (by the number of theme options) if we add anything else to the menu.

Reply:
```
The ++ incrementation happens after the value is accessed 
```
Molly and I both asked about this, so it might be worth mentioning this in a comment?

Reply:
Agreed. Will add!